### PR TITLE
dev-qt/*: add version 6.4.0

### PR DESCRIPTION
**Pull request by Chiitoo**

### CODE REVIEW

**Comment by ionenwks** [comment]
Haven't really looked yet, but diff from 6.3.2 ebuilds for convenience:
```diff
--- dev-qt/qtcharts/qtcharts-6.3.2.ebuild
+++ dev-qt/qtcharts/qtcharts-6.4.0.ebuild
@@ -0,0 +1,18 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit qt6-build
+
+DESCRIPTION="Chart component library for the Qt6 framework"
+
+if [[ ${QT6_BUILD_TYPE} == release ]]; then
+	KEYWORDS="~amd64"
+fi
+
+DEPEND="
+	=dev-qt/qtbase-${PV}*[gui,opengl,widgets]
+	=dev-qt/qtdeclarative-${PV}*
+"
+RDEPEND="${DEPEND}"
--- dev-qt/qtwebengine/qtwebengine-6.3.2.ebuild
+++ dev-qt/qtwebengine/qtwebengine-6.4.0.ebuild
@@ -6,6 +6,6 @@
 PYTHON_COMPAT=( python3_{9,10} )
 PYTHON_REQ_USE="xml(+)"
-CHROMIUM_VER="94.0.4606.126"
-CHROMIUM_PATCHES_VER="104.0.5112.81"
+CHROMIUM_VER="102.0.5005.177"
+CHROMIUM_PATCHES_VER="104.0.5112.102"
 
 inherit check-reqs estack flag-o-matic multiprocessing python-any-r1 qt6-build
@@ -71,4 +71,5 @@
 	x11-libs/libXrender
 	x11-libs/libXScrnSaver
+	x11-libs/libxshmfence:=
 	x11-libs/libXtst
 	alsa? ( media-libs/alsa-lib )
--- dev-qt/qtbase/qtbase-6.3.2.ebuild
+++ dev-qt/qtbase/qtbase-6.4.0.ebuild
@@ -13,6 +13,5 @@
 
 # Qt Modules
-# TODO: Restore/patch xml flag support (seems fixed in 6.9999).
-IUSE="+concurrent +dbus +gui +network +sql opengl +widgets zstd"
+IUSE="+concurrent +dbus +gui +network +sql opengl +widgets +xml zstd"
 REQUIRED_USE="
 	opengl? ( gui )
@@ -93,5 +92,5 @@
 		x11-libs/libSM
 		x11-libs/libX11
-		>=x11-libs/libxcb-1.12:=[xkb]
+		>=x11-libs/libxcb-1.12:=
 		>=x11-libs/libxkbcommon-0.5.0[X]
 		x11-libs/xcb-util-image
@@ -106,17 +105,17 @@
 src_configure() {
 	local mycmakeargs=(
+		-DINSTALL_ARCHDATADIR=${QT6_ARCHDATADIR}
 		-DINSTALL_BINDIR=${QT6_BINDIR}
+		-DINSTALL_DATADIR=${QT6_DATADIR}
+		-DINSTALL_DOCDIR=${QT6_DOCDIR}
+		-DINSTALL_EXAMPLESDIR=${QT6_EXAMPLESDIR}
 		-DINSTALL_INCLUDEDIR=${QT6_HEADERDIR}
 		-DINSTALL_LIBDIR=${QT6_LIBDIR}
-		-DINSTALL_ARCHDATADIR=${QT6_ARCHDATADIR}
-		-DINSTALL_PLUGINSDIR=${QT6_PLUGINDIR}
 		-DINSTALL_LIBEXECDIR=${QT6_LIBEXECDIR}
+		-DINSTALL_MKSPECSDIR=${QT6_ARCHDATADIR}/mkspecs
+		-DINSTALL_PLUGINSDIR=${QT6_PLUGINDIR}
 		-DINSTALL_QMLDIR=${QT6_QMLDIR}
-		-DINSTALL_DATADIR=${QT6_DATADIR}
-		-DINSTALL_DOCDIR=${QT6_DOCDIR}
-		-DINSTALL_TRANSLATIONSDIR=${QT6_TRANSLATIONDIR}
 		-DINSTALL_SYSCONFDIR=${QT6_SYSCONFDIR}
-		-DINSTALL_MKSPECSDIR=${QT6_ARCHDATADIR}/mkspecs
-		-DINSTALL_EXAMPLESDIR=${QT6_EXAMPLESDIR}
+		-DINSTALL_TRANSLATIONSDIR=${QT6_TRANSLATIONDIR}
 		-DQT_FEATURE_androiddeployqt=OFF
 		$(qt_feature concurrent)
@@ -129,4 +128,5 @@
 		$(qt_feature systemd journald)
 		$(qt_feature udev libudev)
+		$(qt_feature xml)
 		$(qt_feature zstd)
 	)
@@ -177,2 +177,9 @@
 	qt6-build_src_configure
 }
+
+src_install() {
+	qt6-build_src_install
+
+	# https://bugs.gentoo.org/863395
+	dosym ../$(get_libdir)/qt6/bin/qmake /usr/bin/qmake6
+}
--- dev-qt/qttools/qttools-6.3.2.ebuild
+++ dev-qt/qttools/qttools-6.4.0.ebuild
@@ -13,8 +13,7 @@
 
 IUSE="
-	assistant designer distancefieldgenerator linguist pixeltool
+	assistant designer distancefieldgenerator +linguist pixeltool
 	qdbus qdoc qtattributionsscanner qtdiag qtplugininfo
 "
-REQUIRED_USE="linguist? ( designer )"
 
 DEPEND="
@@ -33,4 +32,8 @@
 RDEPEND="${DEPEND}"
 
+PATCHES=(
+	"${FILESDIR}/${PN}-6.4.0-clang-15-build.patch" # Bug 873640
+)
+
 src_configure() {
 	local mycmakeargs=(
```

File: dev-qt/qtwebengine/qtwebengine-6.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	system-ffmpeg? ( media-video/ffmpeg:= )
+	system-icu? ( >=dev-libs/icu-69.1:= )
+	widgets? (
+		=dev-qt/qtbase-${PV}*[widgets]
+	)
+"
+DEPEND="${RDEPEND}
+	media-libs/libglvnd
+"
+
```
Comment (ionenwks) [inline dev-qt/qtwebengine/qtwebengine-6.4.0.ebuild:89]: Could you add https://github.com/qt/qtwebengine/commit/81bf140583.patch ? (can be removed in 6.4.1)
It's for https://bugreports.qt.io/browse/QTBUG-107144, led to instant crash whenever I opened github -- other than that 6.4.0 is not giving me issues.

Thanks.

Comment (Chiitoo) [reply dev-qt/qtwebengine/qtwebengine-6.4.0.ebuild:89]: What a fun one, that.

Done.

File: dev-qt/qttools/files/qttools-6.4.0-clang-15-build.patch [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Gentoo Bug: https://bugs.gentoo.org/873640
```
Comment (thesamesam) [inline dev-qt/qttools/files/qttools-6.4.0-clang-15-build.patch:1]: nice headers, thanks for doing these

Comment (Chiitoo) [reply dev-qt/qttools/files/qttools-6.4.0-clang-15-build.patch:1]: For once I didn't forget to.  \ö/

**Pull request closed without merging**